src/Raven.Server/Documents/TimeSeries/TimeSeriesStorage.cs

```diff
@@ -1052,11 +1054,69 @@ private void AddValueInternal(DateTime time, Span<double> values, Span<byte> tag
                 var inRange = timestampDiff < int.MaxValue;
                 if (inRange == false || segment.Append(_context.Allocator, (int)timestampDiff, values, tagSlice, status) == false)
                 {
+                    if (segment.GetLastTimestamp(BaselineDate) == time)
+                    {
+                        TrimSegmentToDate(ref segment, values, time);
+                        segment.Append(_context.Allocator, 0, values, tagSlice, status);
+                        return;
```

Comment:
Aren't we missing `FlushCurrentSegment` here?

Reply:
We already flushed the split segment in `TrimSegmentToDate`, line 1101. Then we initialized it so we can continue adding values without flushing again.

Reply:
Don't you need to call it after the `Append` ?

Reply:
I don't believe so because after the `Append` I'm working on a new (initialized) segment and maybe there are more elements to add in `SplitSegment` so I don't want to flush it yet.